### DISCUSSION TITLE
Upgrading from very old version

### DISCUSSION THREAD

**Discussion by Daniel15** [ohmyzsh/ohmyzsh#10990] [Help]
I haven't updated for a long time. Sorry. My development server at work is currently on rev aaf7fa007fa05c3c2b0b5fdc983494b7442a6196 from March 2017. If I upgrade to the latest version, is there anything I need to be aware of that could completely break anything? I'm using my own custom theme so I won't be surprised if I need to rewrite it, but any other major things? Should I just delete everything including my `.zshrc` and start from scratch?

**Comment by mcornella** [comment]
I tried this on a test container and it looks like there will be a small error at first that you can make go away by running `zsh` again.

Just in case something goes wrong though, be sure to run a new `zsh` session (do not use `exec zsh`), and only when `echo $SHLVL` shows 2 or more, run the upgrade. This is only to make sure that there's no error that prevents loading a working shell session.

Another thing that would work is having multiple tabs open and only perform the update there. After the update, open a new tab, and if everything seems to work correctly then that's all well and done.

There may be some changes in plugins or themes that you use but none that will prevent loading a shell session. Reply to this answer with a list of in-use plugins and theme if you need more details about this.

And as always, if you have an error open a new issue with the details (again, if it's an error that prevents open a new shell, do not close any tabs open and contact again).

**Reply by Daniel15** [reply]
Thanks for the reply!

The only plugins I'm using are git, mercurial and tmux, and I don't even use tmux any more (switched to Eternal Terminal a while back) so I probably don't need that plugin.

I'm using a custom theme:
```
PROMPT='
%F{cyan}%T%  %F{white}%n%F{magenta}@%F{white}%m $fg_bold[green]${PWD/$HOME/~}%b%F{064} $(_dotfiles_scm_info)
%{$fg_bold[yellow]%}%% %{$reset_color%}'
```
(`_dotfiles_scm_info` is provided by my employer, as this is on my work system and we use a customized version of Mercurial)

that looks like this:
![image](https://user-images.githubusercontent.com/91933/173893167-1e20cee1-7a61-446a-8ba8-cf5168cd1b99.png)

**Reply by mcornella** [reply]
In that case this should be the only error you get when updating:
```
/home/user/.oh-my-zsh/lib/directories.zsh:32: command not found: compdef
```
This is because OMZ started with one version and loaded another, but a new zsh session should start without errors.

Again, keep a tab open in case the shell can't start, but that's just being extra careful. Please post back if you successfully updated!
